test/functional/android/remote_fs_tests.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import os
+import random
+import unittest
+from zipfile import ZipFile
+
+from appium import webdriver
+import desired_capabilities
+
+
+class RemoteFsTests(unittest.TestCase):
+    def setUp(self):
+        desired_caps = desired_capabilities.get_desired_capabilities('ApiDemos-debug.apk')
+        self.driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps)
+
+    def tearDown(self):
+        self.driver.quit()
+
+        # remove zipped file from `test_pull_folder`
+        if hasattr(self, 'zipfilename') and os.path.isfile(self.zipfilename):
+            os.remove(self.zipfilename)
+
+    def test_push_pull_file(self):
+        path = '/data/local/tmp/test_push_file.txt'
+        data = b'This is the contents of the file to push to the device.'
+
+        self.driver.push_file(path, base64.b64encode(data).decode('utf-8'))
+        data_ret = base64.b64decode(self.driver.pull_file(path))
+
+        self.assertEqual(data, data_ret)
+
+    def test_pull_folder(self):
+        string_data = b'random string data %d' % random.randint(0, 1000)
+        path = '/data/local/tmp'
+
+        self.driver.push_file(path + '/1.txt', base64.b64encode(string_data).decode('utf-8'))
+        self.driver.push_file(path + '/2.txt', base64.b64encode(string_data).decode('utf-8'))
+
+        folder = self.driver.pull_folder(path)
+
+        # python doesn't have any functionality for unzipping streams
```

Comment:
how about StringIO?

Reply:
I didn't know StringIO, it's super cool.
But for here, I couldn't create zipfile object(here is `myzip`) with StringIO object.

Reply:
hm, should be possible: https://stackoverflow.com/questions/10908877/extracting-a-zipfile-to-memory

Reply:
https://stackoverflow.com/questions/5710867/downloading-and-unzipping-a-zip-file-without-writing-to-disk/42119017

Reply:
Thanks so much, I did.
(I didn't take care of decoding for zipfile data...)